### PR TITLE
feat: calculate non-profiled consumption per energy supplier

### PR DESCRIPTION
**Pull request by JesperGomes**
## References

* #199

### CODE REVIEW

File: source/databricks/package/calculation_input.py [hunk 1]
```diff
@@ -47,7 +47,10 @@ def get_metering_point_periods_df(
             col(Colname.to_date).isNull()
             | (col(Colname.to_date) > period_start_datetime)
         )
-        .where(col(Colname.metering_point_type) == MeteringPointType.production.value)
+        .where(
+            (col(Colname.metering_point_type) == MeteringPointType.production.value)
+            | (col(Colname.metering_point_type) == MeteringPointType.consumption.value)
+        )
```
Comment (BjarkeMeier) [inline source/databricks/package/calculation_input.py:53]: Should this be completely removed? What about exchange?

File: source/databricks/package/calculator_job.py [hunk 2]
```diff
@@ -101,7 +101,8 @@ def _start_calculator(spark: SparkSession, args: CalculatorArgs) -> None:
     )
 
     (
-        result_df,
+        non_profiled_consumption_per_ga_and_es,
```
Comment (BjarkeMeier) [inline source/databricks/package/calculator_job.py:104]: Is it about time to create a (data)class to hold the output? Perhaps in another PR.

File: source/databricks/package/calculator_job.py [hunk 3]
```diff
@@ -132,17 +133,27 @@ def _start_calculator(spark: SparkSession, args: CalculatorArgs) -> None:
...
     )
 
+    path = f"{args.process_results_path}/batch_id={args.batch_id}/result"
+
     # First repartition to co-locate all rows for a grid area on a single executor.
     # This ensures that only one file is being written/created for each grid area
     # When writing/creating the files. The partition by creates a folder for each grid area.
-    # result/
+
+    # Total production
```
Comment (BjarkeMeier) [inline source/databricks/package/calculator_job.py:142]: Is it time to refactor this out of this ever-growing file?

File: source/databricks/package/calculator_job.py [hunk 4]
```diff
@@ -182,7 +193,7 @@ def _start(command_line_args: list[str]) -> None:
 
     if islocked(args.data_storage_account_name, args.data_storage_account_key):
         log("Exiting because storage is locked due to data migrations running.")
-        sys.exit(3)
+        exit(3)
```
Comment (BjarkeMeier) [inline source/databricks/package/calculator_job.py:196]: Funny. Lasse made the opposite change as this solution failed in Databricks. However, he probably didn't add a unit test ensuring it. So now you might be re-introducing the same error...

File: source/databricks/tests/integration/calculator/calculation_input/test_get_metering_point_periods_df.py [hunk 5]
```diff
@@ -186,12 +188,12 @@ def test__when_type_is_production__returns_metering_point_period(
     assert raw_master_basis_data.count() == 1
 
 
-def test__when_type_is_not_E18__does_not_returns_metering_point_period(
-        batch_grid_areas_df,
-        metering_points_periods_df_factory,
+def test__when_type_is_not_E17_or_E18__does_not_returns_metering_point_period(
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/calculation_input/test_get_metering_point_periods_df.py:191]: nit: 
```suggestion
def test__when_type_is_not_E17_or_E18__does_not_return_metering_point_period(
```

File: source/databricks/package/calculator_job.py (hunk 4 above)
Comment (JesperGomes) [reply source/databricks/package/calculator_job.py:196]: I think PyLance did this automatically. I can revert it.

File: source/databricks/package/calculator_job.py (hunk 2 above)
Comment (JesperGomes) [reply source/databricks/package/calculator_job.py:104]: Yes, or maybe do it with the next calculation we do!?!

File: source/databricks/package/calculator_job.py (hunk 3 above)
Comment (JesperGomes) [reply source/databricks/package/calculator_job.py:142]: Yes, agree. I would prefer to do it a separate task. I can make one on the story.

File: source/databricks/package/calculation_input.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/package/calculation_input.py:53]: I think we can remove it. I will try it.

File: source/databricks/tests/integration/calculator/calculation_input/test_get_metering_point_periods_df.py (hunk 5 above)
Comment (JesperGomes) [reply source/databricks/tests/integration/calculator/calculation_input/test_get_metering_point_periods_df.py:191]: I think we can remove this test. It's not a problem that we return periods that includes other types like E20

File: source/databricks/package/calculation_input.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/package/calculation_input.py]: I have now removed it

File: source/databricks/package/balance_fixing.py [hunk 6]
```diff
@@ -53,34 +53,88 @@ def calculate_balance_fixing(
         metering_points_periods_df, period_start_datetime, period_end_datetime
     )
 
+    metadata_fake = Metadata("1", "1", "1", "1")
+
     results = {}
     results[ResultKeyName.aggregation_base_dataframe] = enriched_time_series_point_df
-    metadata_fake = Metadata("1", "1", "1", "1")
+    results[ResultKeyName.non_profiled_consumption] = agg_steps.aggregate_consumption(
```
Comment (BjarkeMeier) [inline source/databricks/package/balance_fixing.py:60]: Perhaps?
```suggestion
    results[ResultKeyName.non_profiled_consumption] = agg_steps.aggregate_non_profiled_consumption(
```

File: source/databricks/package/balance_fixing.py [hunk 7]
```diff
@@ -53,34 +53,88 @@ def calculate_balance_fixing(
...
+        results, metadata_fake
+    )
+
+    # Non-profiled consumption per energy supplier
+    consumption_per_ga_and_es = agg_steps.aggregate_non_profiled_consumption_ga_es(
```
Comment (BjarkeMeier) [inline source/databricks/package/balance_fixing.py:65]: The dependency is hard to tell

File: source/databricks/package/balance_fixing.py [hunk 8]
```diff
@@ -53,34 +53,88 @@ def calculate_balance_fixing(
...
+        results, metadata_fake
+    )
+
+    consumption_per_ga_and_es = _prepare_result_for_output(
+        consumption_per_ga_and_es,
+        TimeSeriesType.NON_PROFILED_CONSUMPTION,
+        ResultGrouping.PER_ENERGY_SUPPLIER,
```
Comment (BjarkeMeier) [inline source/databricks/package/balance_fixing.py:72]: The other output related stuff is currently in calculator_job, what is the reason to keep something here?

File: source/databricks/package/balance_fixing.py (hunk 7 above)
Comment (BjarkeMeier) [inline source/databricks/package/balance_fixing.py:65]: What does the "ga" in the names tell?

File: source/databricks/package/constants/time_series_type.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+
+
+class TimeSeriesType(Enum):
```
Comment (BjarkeMeier) [inline source/databricks/package/constants/time_series_type.py:18]: Nice :-)

File: source/databricks/tests/integration/calculator/test_calculator_job.py [hunk 10]
```diff
@@ -168,21 +171,32 @@ def test__result_is_generated_for_requested_grid_areas(
     worker_id,
     executed_calculation_job,
 ):
+    # Arrange
     data_lake_path = f"{data_lake_path}/{worker_id}"
 
+    expected_ga_gln_type = [
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/test_calculator_job.py:177]: Do we like having such a constantly changing test?

**Review by BjarkeMeier — APPROVED** [review]

File: source/databricks/package/balance_fixing.py (hunk 8 above)
Comment (JesperGomes) [reply source/databricks/package/balance_fixing.py:72]: Yes, they should be closer to eachother - not necessarily in calculator_job though. I will make a task for considering these refactoring for separate PR

File: source/databricks/package/balance_fixing.py (hunk 7 above)
Comment (JesperGomes) [reply source/databricks/package/balance_fixing.py:65]: it's aligned with what PO/SME normally writes

File: source/databricks/tests/integration/calculator/test_calculator_job.py (hunk 10 above)
Comment (JesperGomes) [reply source/databricks/tests/integration/calculator/test_calculator_job.py:177]: No really

File: source/databricks/package/balance_fixing.py (hunk 7 above)
Comment (JesperGomes) [reply source/databricks/package/balance_fixing.py:65]: Agree - candidate to a separate PR

**Pull request merged**